src/modules/openPopup.js

```diff
@@ -0,0 +1,134 @@
+    fetch(`https://pokeapi.co/api/v2/pokemon/${id}`)
+      .then((response) => response.json())
+      .then((pokemonInfo) => {
+        renderPkmnPopup(pokemonInfo);
+      });
+  }
+
+  const displayComments = async (id, commentDiv) => {
+    const pokemonComments = await getComments(id);
+    pokemonComments.forEach((comment) => {
+      const comDiv = document.createElement('div');
+      comDiv.classList.add('single-comment');
+      const user = document.createElement('p');
+      user.innerHTML = `${comment.username}: `;
+      const userCom = document.createElement('p');
+      userCom.innerHTML = comment.comment;
+      comDiv.append(user, userCom);
+      commentDiv.append(comDiv);
+    });
+  };
+
+  const countComments = async (id, counterP) => {
+    const pokemonComments = await getComments(id);
+    const commentsCounter = pokemonComments.length;
+    counterP.innerHTML = commentsCounter;
+  };
+
+  const removeComments = () => {
+    const comsDiv = document.getElementById('all-comments');
+    comsDiv.innerHTML = '';
+  };
+
+  function renderPkmnPopup(pokemonInfo) {
+    const PopupPokemonContainer = document.getElementById('container');
+    const pokePopup = document.createElement('div');
+    pokePopup.classList.add('popup-card');
+    pokePopup.setAttribute('id', 'popup');
+
+    const popupDetails = document.createElement('div');
+    popupDetails.classList.add('popup-details');
+
+    const pokeDetails = document.createElement('div');
+    pokeDetails.classList.add('popup-poke-details');
+
+    createImg(pokemonInfo.id, pokeDetails);
+
+    const closeCommentBtn = document.createElement('button');
+    closeCommentBtn.classList.add('close-comment-btn');
+    closeCommentBtn.innerHTML = 'Close';
+    closeCommentBtn.addEventListener('click', () => {
+      const popup = document.getElementById('popup');
+      popup.remove();
+    });
+
+    const pokeName = document.createElement('h4');
+    pokeName.innerText = pokemonInfo.name;
+
+    const pokeNumber = document.createElement('p');
+    pokeNumber.innerText = `#${pokemonInfo.id}`;
+
+    const likePokemon = document.createElement('div');
+    likePokemon.className = 'like-pokemon';
+
+    const likeIcon = document.createElement('i');
+    likeIcon.className = 'far fa-heart fa-lg';
+    likeIcon.setAttribute('id', `like${pokemonInfo.id}`);
+
+    const likeCounter = document.createElement('p');
+    likeCounter.classList.add('like-counter');
+
+    likeCountes(pokemonInfo.id, likeCounter);
+
+    const pokeTypes = document.createElement('ul');
+
+    createPokeType(pokemonInfo.types, pokeTypes);
+
+    const commentsdiv = document.createElement('div');
+    commentsdiv.classList.add('comments-div');
+    commentsdiv.innerHTML = `
+        <div id="all-comments" class="comments">
+        </div>
+        <form id="comment-form">
+            <div class="counter-div">
+              <p>Comments: </p><p id="comments-counter"></p>
+            </div>
```

Comment:
- [ ] Currently, the comments counter is returning `undefined` when there is no comment. Please make sure to address this issue and display the correct value when no comment is present.

Reply:
Thank you for your comment, now we fixed this bug by checking the count of comments.